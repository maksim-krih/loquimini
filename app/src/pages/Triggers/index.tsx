import React, { FC, useState } from "react";
import { Button, Input } from "antd";
import { IProps } from "./types";
import { useStyles } from "./styles";
import Api, { AuthService } from "../../services";
import { useHistory } from "react-router-dom";
import { RouterPaths } from "../../consts";

const Trigger: FC<IProps> = (props: IProps) => {
  const classes = useStyles();
  const history = useHistory();
  const [loading, setLoading] = useState(false);

  const generateReceipts = async () => {
    setLoading(true);
    Api.Receipt.generateReceipts()
    .then((result: any) => {
      
    })
    .catch((e: any) => {

    })
    .finally(() => {
      setLoading(false);
    });
  };

  return (
    <div className={classes.container}>
      <Button 
        id="generate-receipts"
        onClick={generateReceipts} 
        style={{ borderRadius: 5 }} 
        type="primary"
        loading={loading}
      >
        Generate Receipts
      </Button>
    </div>
  );
}

export default Trigger;