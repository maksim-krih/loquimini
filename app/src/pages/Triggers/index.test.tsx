import Triggers from ".";
import { shallow } from "enzyme";

const setUp = () => shallow(<Triggers />);

describe("Triggers", () => {
  it("should render Triggers Component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });

  it("should Generate receipts Triggers Component", () => {
    const component = setUp();
    const button = component.find("#generate-receipts");
    button.simulate("click");
  });
});