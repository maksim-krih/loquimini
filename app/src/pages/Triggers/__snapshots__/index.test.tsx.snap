// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Triggers should render Triggers Component 1`] = `
<div
  className="container-0-2-1"
>
  <Button
    id="generate-receipts"
    loading={false}
    onClick={[Function]}
    style={
      Object {
        "borderRadius": 5,
      }
    }
    type="primary"
  >
    Generate Receipts
  </Button>
</div>
`;
